File: Scanner/DataBaseFramework/Applications/SQLServerScriptRunner - Copy/Classes/SQLScript.vb
Imports System.IO
Imports System.Text
Imports System.Text.RegularExpressions

Namespace SQLServerScripting

    Public Class SQLScript
        Private _scriptBatchContents As Collections.Queue
        Private _filename As String
        Private _filePathName As String

        Public Sub New(ByVal fileName As String)
            Try
                'gets the filename
                Dim file As New FileInfo(fileName)
                Me._filename = file.Name
                Me._filePathName = file.FullName
                GetFileContents(Me._filePathName)
            Catch ex As Exception
                Throw ex
            End Try
        End Sub

        Public ReadOnly Property ScriptBatchContents() As Queue
            Get
                Return Me._scriptBatchContents
            End Get
        End Property

        Public ReadOnly Property FileName() As String
            Get
                Return Me._filename
            End Get
        End Property

        Private Sub GetFileContents(ByVal fileName As String)
            Try
                If Me._scriptBatchContents Is Nothing Then
                    Me._scriptBatchContents = New Collections.Queue
                End If

                Dim fileReader As New StreamReader(fileName)
                Dim currentScript As New StringBuilder
                Dim currentLine As String
                Dim rexLn As New Regex("--", RegexOptions.Singleline)
                Dim rexGo As New Regex("^go$", RegexOptions.IgnoreCase)
                Dim insertionFlag As Boolean = False

                currentLine = fileReader.ReadLine

                While (Not currentLine Is Nothing)
                    If Not rexLn.Match(currentLine).Success Then
                        If rexGo.Match(currentLine).Success Then
                            _scriptBatchContents.Enqueue(currentScript.ToString)
                            insertionFlag = True
                            currentScript = New StringBuilder
                        Else
                            currentScript.Append(currentLine)
                            currentScript.Append(vbCrLf)
                        End If
                    End If
                    currentLine = fileReader.ReadLine
                End While

                If Not insertionFlag Then
                    _scriptBatchContents.Enqueue(currentScript.ToString)
                End If
            Catch ex As Exception
                Throw ex
            End Try
        End Sub

    End Class
End Namespace

File: Scanner/DataBaseFramework/Applications/SQLServerScriptRunner/Classes/SQLScript.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using System.Diagnostics;
using System.Data;
using System.Xml.Linq;
using Microsoft.VisualBasic;
using System.Collections;
using System.Linq;
// End of VB project level imports

using System.IO;
using System.Text;
using System.Text.RegularExpressions;


namespace SQLServerScriptRunner
{
	namespace SQLServerScripting
	{
		
		public class SQLScript
		{
			private System.Collections.Queue _scriptBatchContents;
			private string _filename;
			private string _filePathName;
			
			public SQLScript(string fileName)
			{
				try
				{
					//gets the filename
					FileInfo file = new FileInfo(fileName);
					this._filename = file.Name;
					this._filePathName = file.FullName;
					GetFileContents(this._filePathName);
				}
				catch (Exception ex)
				{
					throw (ex);
				}
			}
			
public Queue ScriptBatchContents
			{
				get
				{
					return this._scriptBatchContents;
				}
			}
			
public string FileName
			{
				get
				{
					return this._filename;
				}
			}
			
			private void GetFileContents(string fileName)
			{
				try
				{
					if (this._scriptBatchContents == null)
					{
						this._scriptBatchContents = new System.Collections.Queue();
					}
					
					StreamReader fileReader = new StreamReader(fileName);
					StringBuilder currentScript = new StringBuilder();
					string currentLine = "";
					Regex rexLn = new Regex("--", RegexOptions.Singleline);
					Regex rexGo = new Regex("^go$", RegexOptions.IgnoreCase);
					bool insertionFlag = false;
					
					currentLine = fileReader.ReadLine();
					
					while (!(currentLine == null))
					{
						if (!rexLn.Match(currentLine).Success)
						{
							if (rexGo.Match(currentLine).Success)
							{
								_scriptBatchContents.Enqueue(currentScript.ToString());
								insertionFlag = true;
								currentScript = new StringBuilder();
							}
							else
							{
								currentScript.Append(currentLine);
								currentScript.Append("\r\n");
							}
						}
						currentLine = fileReader.ReadLine();
					}
					
					if (!insertionFlag)
					{
						_scriptBatchContents.Enqueue(currentScript.ToString());
					}
				}
				catch (Exception ex)
				{
					throw (ex);
				}
			}
			
		}
	}
}
